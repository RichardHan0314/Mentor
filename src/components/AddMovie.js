import React from 'react';
import styled from 'styled-components';

const AddMovie = ({handleShow}) => {
    return (
        <Button onClick={handleShow}>+ Add Movie</Button>
    );
};

const Button = styled.button`
    height: 50px;
    padding: 0px 20px;
    color: #f65261;
    font-size: 20px;
    text-align: center;
    background: hsl(0 0% 33% / 0.7);
    border: 0;
    border-radius: 4px;
    text-transform: uppercase;
    cursor: pointer;
`;

export default AddMovie;
