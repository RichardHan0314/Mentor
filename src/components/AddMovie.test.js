import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import AddMovie from './AddMovie';

describe('AddMovie component', () => {
    it('Render AddMovie snapshot', () => {
        const { asFragment } = render(<AddMovie />);
        expect(asFragment(<AddMovie />)).toMatchSnapshot();
    });

    it('Render AddMovie correctly', () => {
        render(<AddMovie />);
        screen.debug();
        expect(screen.getByText('+ Add Movie')).toBeInTheDocument();
    });

    it('calls the onClick callback handler', () => {
        const handleShow = jest.fn();

        render(<AddMovie handleShow={handleShow}/>);

        const AddMovieBtn = screen.getByRole('button');
        fireEvent.click(AddMovieBtn);
        expect(handleShow).toHaveBeenCalledTimes(1);
    });
});
