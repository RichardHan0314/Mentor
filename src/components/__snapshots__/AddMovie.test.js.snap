// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovie component Render AddMovie snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm lhlTFD"
  >
    + Add Movie
  </button>
</DocumentFragment>
`;

exports[`AddMovie component render AddMovie snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm lhlTFD"
  >
    + Add Movie
  </button>
</DocumentFragment>
`;
